test/FileS3Loader.js

```diff
@@ -23,13 +23,13 @@ export class FileS3Loader {
         'helix-code-bus': path.resolve(__testdir, 'fixtures', 'code'),
       },
       statusCodeOverrides: {},
-      rewrites: {},
+      rewrites: [],
       headerOverride: {},
     });
   }
 
   rewrite(fileName, dst) {
-    this.rewrites[fileName] = dst;
+    this.rewrites.push((key) => (key.endsWith(fileName) ? dst : null));
```

Comment:
make `rewrite` a bit more context-sensitive (we can't load 2 files called `metadata.json` in a test when we just have the basename)